src/object_store/src/object/s3.rs

```diff
@@ -617,8 +617,8 @@ impl S3ObjectStore {
         let server = server.strip_prefix("minio://").unwrap();
         let (access_key_id, rest) = server.split_once(':').unwrap();
         let (secret_access_key, rest) = rest.split_once('@').unwrap();
-        let (address, bucket) = rest.split_once('/').unwrap();
-
+        let (address, bucket) = rest.split_once('#').unwrap();
```

Comment:
This change breaks compatibility meaning that deployment running prior to this PR will break. Can we make it compatible? For example, we can keep the original format for the url for http and use the following one for https:
```
hummock+minio_https://hummockadmin:hummockadmin@127.0.0.1:9301/hummock001
```

Reply:
`hummock+minio_https://hummockadmin:hummockadmin@127.0.0.1:9301/hummock001` also break the compatibility as we will `server.strip_prefix("minio://").unwrap();`.

Reply:
I think regex can be used to avoid compatibility issues, let me try it,

Reply:
We can match `minio_https` first before `minio`. regex also works.

Reply:
Yeap, no matter which modification method is used, corresponding changes are required on the operator side.